models/Duplication_model.slim

```diff
@@ -0,0 +1,167 @@
+    initializeMutationType("m2", 0.5, "f", 0.0);     // Duplication marker
+    m2.color = "red";
+    m2.convertToSubstitution = F;
+
+    // Define genomic element
+    initializeGenomicElementType("g1", m1, 1.0, mmJukesCantor(1e-7));
+    initializeGenomicElement(g1, 0, L - 1);
+
+    // Set recombination rate
+    initializeRecombinationRate(1e-8);
+}
+
+
+1 early() {
+    defineConstant("simID", getSeed());
+    sim.addSubpop("p1", N);  // Initialize population
+}
+
+recombination() {
+    gm1 = genome1.containsMarkerMutation(m2, DUP_START);
+    gm2 = genome2.containsMarkerMutation(m2, DUP_START);
+    if (!(gm1 | gm2)) {
+        // homozygote non-duplicated
+        // Filter breakpoints that exceed the recombination limit
+        breakpoints = breakpoints[breakpoints < DUP_START];
+
+        // If no breakpoints remain after filtering, return F (no recombination)
+        if (length(breakpoints) == 0) {
+            return F;
+        }
+
+        // Otherwise, allow the filtered breakpoints to proceed
+        return T;
+    }
+    if (gm1 & gm2) {
+        //homozygot duplication
+        return F;
+    }
+        else {
+        // heterozygote duplicated: resample to get an even # of breakpoints
+        // Filter breakpoints that exceed the recombination limit
+        breakpoints = breakpoints[breakpoints < DUP_START];
+
+        // If no breakpoints remain after filtering, return F (no recombination)
+        if (length(breakpoints) == 0) {
+            return F;
+        }
+
+        // Otherwise, allow the filtered breakpoints to proceed
+        return T;
+    }
+}
+
+2:999 late(){
+    for (genome in p1.genomes) {
+        // Check if the genome does NOT have the marker mutation m2 at DUP_START
+        allMuts = 0;
+        if (!(genome.containsMarkerMutation(m2, DUP_START))) {
+
+allMuts = genome.mutations;
+mutsToRemove = allMuts[allMuts.position > DUP_START];
+sim.subpopulations.genomes.removeMutations(mutsToRemove);
+
+}
+}
+
+}
+
+
+1000 late() {
+    sim.outputFull(tempdir() + "burnin_" + simID + ".txt");  // Save state after burn-in
+    catn("Burn-in phase complete. State saved.");
+}
+
+
+1001 late() {
+    // Introduce duplication mutation in one individual
+    duplicated = sample(p1.genomes, 1);
+    duplicated.addNewDrawnMutation(m2, DUP_START);
+    catn("Duplication mutation introduced at generation 1001.");
+
+    // Copy mutations from the original region to the duplicated region
+    originalRegion = c(DUP_START - DUP_LENGTH, DUP_START - 1);
+
+    // Identify mutations in the original region
+    mutsToCopy = duplicated.mutations[duplicated.mutations.position >= originalRegion[0] & duplicated.mutations.position <= originalRegion[1]];
+
+    // Duplicate these mutations to the new region
+    for (mut in mutsToCopy) {
+        newPosition = mut.position + DUP_LENGTH;
+        duplicated.addNewDrawnMutation(mut.mutationType, newPosition, p1, mut.nucleotide);
```

Comment:
OK; I'd suggest using `addNewMutation()` instead, though, and passing in `mut.selectionCoeff`.  In this model the mutations are all neutral so it doesn't matter, but using `addNewMutation()` would be a more general solution that would work even if the mutations were not all neutral.